### PR TITLE
DAOS-6891 control: Clean up JSON output for errors

### PR DESCRIPTION
**Pull request by mjmac**
Rather than handling errors piecemeal, centralize the
decision about whether or not an error should be formatted
as JSON.

### CODE REVIEW

**Review by tanabarr — APPROVED** [review]

File: src/control/cmd/dmg/pool.go [hunk 1]
```diff
@@ -213,7 +213,7 @@ func (cmd *PoolDestroyCmd) Execute(args []string) error {
 	err := control.PoolDestroy(ctx, cmd.ctlInvoker, req)
 
 	if cmd.jsonOutputEnabled() {
```
Comment (tanabarr) [inline src/control/cmd/dmg/pool.go:215]: do we need this check?

**Review by tanabarr — DISMISSED** [review]

File: src/control/cmd/dmg/pool.go (hunk 1 above)
Comment (mjmac) [reply src/control/cmd/dmg/pool.go]: Nope, good catch. I had to push again after getting the fixes for mpiio/romio from master, so I've just removed it.

**Review by tanabarr — APPROVED** [review]

**Comment by mjmac** [comment]
Known failure:
- [DAOS-6932](https://jira.hpdd.intel.com/browse/DAOS-6932)

**Review by kjacque — COMMENTED** [review]
LGTM. Sorry for the delay on reviewing.

**Pull request merged**